src/main/java/team4/footwithme/member/service/MemberServiceImpl.java

```diff
@@ -55,9 +62,65 @@ public LoginResponse login(LoginServiceRequest serviceRequest, HttpServletRespon
         redisTemplate.opsForValue().set(member.getEmail(), tokenResponse.refreshToken(), tokenResponse.refreshTokenExpirationTime(), TimeUnit.MICROSECONDS);
         // Redis에 RefreshToken 저장
 
-        cookieService.setHeader(response, tokenResponse.refreshToken());
-        // 쿠키에 refreshToken 저장
-
         return LoginResponse.from(tokenResponse);
     }
+
+    @Override
+    @Transactional
+    public String logout(HttpServletRequest request) {
+        String accessToken = jwtTokenUtil.resolveToken(request);
+        String email = jwtTokenUtil.getEmailFromToken(accessToken);
+
+        jwtTokenUtil.tokenValidation(accessToken);
+
+        if(redisTemplate.opsForValue().get(email) != null){
+            redisTemplate.delete(email);
+        }
+
+        long expiration = jwtTokenUtil.getExpiration(accessToken);
+        redisTemplate.opsForValue().set(accessToken, "logout", expiration, TimeUnit.MICROSECONDS);
+
+        return "Success Logout";
+    }
+
+    @Override
+    public TokenResponse reissue(HttpServletRequest request, String refreshToken) {
+        if(refreshToken.isEmpty()){
```

Comment:
refreshToken이 null일 경우 isEmpty는 NullPointerException을 발생시킵니다.
아마 다른 곳에서 처리하셨을 수 있지만, 저는 isEmpty를 사용할 때 null체크를 같이 하는 편이에요.
```suggestion
        if(refreshToken == null || refreshToken.isEmpty()){
```

Reply:
아하 그러게요 null 값 체크가 안됐을 수 있을 거 같아요 고쳐보겠습니다!